PhysicsTools/MVATrainer/src/TrainProcessor.cc

```diff
@@ -154,8 +154,8 @@ TrainProcessor *ProcessRegistry<TrainProcessor, AtomicId,
 	if (!result) {
 		// try to load the shared library and retry
 		try {
-			delete TrainProcessor::PluginFactory::get()->create(
-				std::string("TrainProcessor/") + name);
+			std::unique_ptr<PhysicsTools::TrainProcessor::Dummy> tmp{TrainProcessor::PluginFactory::get()->create(
+				std::string("TrainProcessor/") + name)};
 			result = ProcessRegistry::create(name, id, trainer);
 		} catch(const cms::Exception &e) {
```

Comment:
In principle the try-catch here is unnecessary, and the behavior could be preserved by using `PluginFactory::tryToCreate()`. On the other hand, the comment below says
```
// caller will have to deal with the null pointer
// in principle this will just give a slightly more
// descriptive error message (and will rethrow anyhow)
```
but there is no rethrow. So I'm not sure if this code should throw (keep the `create()` but remove try-catch as unnecessary) or not (change to `tryToCreate()`).